subsys/mgmt/mcumgr/zephyr_grp/basic_mgmt.c

```diff
@@ -33,6 +36,89 @@ int storage_erase(void)
 	return rc;
 }
 
+
+#if defined(CONFIG_MCUMGR_GRP_BASIC_CMD_IMAGE_LIST)
+static int image_status(int id, char *buffer, ssize_t len)
+{
+	const struct flash_area *fa;
+	struct image_header hdr;
+	struct boot_swap_state bss;
+	int rc = flash_area_open(id, &fa);
+
+	if (rc < 0) {
+		LOG_ERR("Flash area %d open failed", id);
+		return -1;
+	}
+
+	rc = flash_area_read(fa, 0, &hdr, sizeof(hdr));
+	if (rc < 0) {
+		flash_area_close(fa);
+		LOG_ERR("Flash area %d read failed", id);
+		return -1;
+	}
+
+	rc = boot_read_swap_state(fa, &bss);
+	flash_area_close(fa);
+	if (rc < 0) {
+		LOG_ERR("Boot swap state %d read failed", id);
+		return -1;
+	}
+	if (hdr.ih_magic == IMAGE_MAGIC) {
+		snprintf(buffer, len, "ver=%d.%d.%d.%d%s", hdr.ih_ver.iv_major,
+			 hdr.ih_ver.iv_minor, hdr.ih_ver.iv_revision,
+			 hdr.ih_ver.iv_build_num,
+			 bss.copy_done == BOOT_FLAG_SET ? ",copy_done" : "");
+		return 1;
+	}
+	/* Not an application, not an error */
+	return 0;
+}
+
+/* Only defined images are probed */
+#define IMAGE_LIST_ITEM(n, node_label)					\
+	COND_CODE_1(DT_HAS_FIXED_PARTITION_LABEL(node_label),		\
+		({							\
+			.num = n,					\
+			.fa_id = FLASH_AREA_ID(node_label),		\
+		},), ())
+
+static int image_list(struct mgmt_ctxt *ctxt)
+{
+	static const struct {
+		int num; int fa_id;
+	} images[] = {
+		IMAGE_LIST_ITEM(1, image_0)
+		IMAGE_LIST_ITEM(2, image_1)
+		IMAGE_LIST_ITEM(3, image_2)
```

Comment:
this is hardcoded for 2 image pairs. W need support at last most common 1 image pair as well.

Reply:
I am not user what you mean, this includes image-0 (mcuboot slot-0), image-1 (mcuboot slot-1), and two next labels may be translated to whatever pair you want.

Reply:
I missed definition above